### PR TITLE
Subprocess output returning map variable

### PR DESCRIPTION
**Pull request by nadberezny**
Allow subprocess outputs to return value.
Subprocess can behave as MapVariable and pass fields to process.

### CODE REVIEW

File: engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/canonicalgraph/ProcessNodesRewriter.scala [hunk 1]
```diff
@@ -144,7 +144,13 @@ trait ExpressionRewriter {
       case n: Source =>
         n.copy(
           ref = n.ref.copy(parameters = rewriteParameters(n.ref.parameters)))
-      case _: BranchEndData | _: Split | _: SubprocessInputDefinition | _: SubprocessOutputDefinition | _: SubprocessOutput => data
+      case n: SubprocessOutputDefinition =>
+        n.copy(
```
Comment (lciolecki) [inline engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/canonicalgraph/ProcessNodesRewriter.scala]: Ugly formatting.. Better will be in one line?

File: engine/flink/process/src/test/scala/pl/touk/nussknacker/engine/process/FlinkStreamingProcessRegistrarKafkaSpec.scala [hunk 2]
```diff
@@ -47,7 +46,7 @@ class FlinkStreamingProcessRegistrarKafkaSpec
         .processorEnd("service", "logService", "all" -> "#input"))
     )
 
-    ProcessTestHelpers.processInvoker.invokeWithKafka(
+    helpers.ProcessTestHelpers.processInvoker.invokeWithKafka(
```
Comment (gadomsky) [inline engine/flink/process/src/test/scala/pl/touk/nussknacker/engine/process/FlinkStreamingProcessRegistrarKafkaSpec.scala]: do we need that change actually?

Comment (nadberezny) [reply engine/flink/process/src/test/scala/pl/touk/nussknacker/engine/process/FlinkStreamingProcessRegistrarKafkaSpec.scala]: we don't but it removes compiler warning

File: engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/canonicalgraph/ProcessNodesRewriter.scala (hunk 1 above)
Comment (nadberezny) [reply engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/canonicalgraph/ProcessNodesRewriter.scala]: Ugliness is a matter of tase probably :)
but back to this match clasue - I've kept the existing code formatting, for sake of style consistency

**Pull request merged**